pkg/tsdb/postgres/postgres.go

```diff
@@ -56,15 +57,124 @@ func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndp
 	return endpoint, err
 }
 
+func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
+	logger := log.New("tsdb.postgres")
+	logger.Debug("Creating Postgres query endpoint")
+	endpoint, err := databaseConnection(datasource, logger)
+	return endpoint, err
+}
+
 // escape single quotes and backslashes in Postgres connection string parameters.
 func escape(input string) string {
 	return strings.ReplaceAll(strings.ReplaceAll(input, `\`, `\\`), "'", `\'`)
 }
 
+func writeConnectionFile(
+	ds *models.DataSource, fileContent string, currentPath string, certFileName string, jsonFieldName string) error {
+	var generatedFilePath string
+	if ssljs, ok := ds.JsonData.CheckGet(jsonFieldName); ok {
+		generatedFilePath = ssljs.MustString("")
+	}
+	if fileContent != "" {
+		if generatedFilePath == "" {
+			generatedFilePath = filepath.Join(currentPath, certFileName)
+		}
+
+		if err := ioutil.WriteFile(generatedFilePath, []byte(fileContent), 0700); err != nil {
+			return err
+		}
+		ds.JsonData.Set(jsonFieldName, generatedFilePath)
+	} else {
+		if generatedFilePath != "" {
+			if _, err := os.Stat(generatedFilePath); err == nil {
+				if err := os.Remove(generatedFilePath); err != nil {
+					return err
+				}
+			}
+		}
+		ds.JsonData.Set(jsonFieldName, "")
+	}
+	return nil
+}
+
+func writeConnectionFiles(ds *models.DataSource, logger log.Logger) error {
+	sslMode := strings.TrimSpace(strings.ToLower(ds.JsonData.Get("sslmode").MustString("verify-full")))
+	decrypted := ds.SecureJsonData.Decrypt()
+	tlsCACert := decrypted["tlsCACert"]
+	tlsClientCert := decrypted["tlsClientCert"]
+	tlsClientKey := decrypted["tlsClientKey"]
+
+	if sslMode == "disable" {
+		return nil
+	}
+
+	// create folder
+	currentPath, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	var mutex = &sync.Mutex{}
+
+	currentPath = filepath.Join(currentPath, ds.Uid+"generatedSSLCerts")
+	if _, err := os.Stat(currentPath); os.IsNotExist(err) {
+		mutex.Lock()
+		err = os.Mkdir(currentPath, 0600)
```

Comment:
```suggestion
		err = os.Mkdir(currentPath, 0700)
```
the directory should have execute permission otherwise it fails to access the directory contents.